addons/sale_timesheet/migrations/11.0.1.0/openupgrade_analysis_work.txt

```diff
@@ -0,0 +1,64 @@
+---Fields in module 'sale_timesheet'---
+sale_timesheet / account.analytic.line    / timesheet_invoice_id (many2one): NEW relation: account.invoice
+# Done: recomputed by calling _compute_timesheet_revenue method
+
+sale_timesheet / account.analytic.line    / timesheet_invoice_type (selection): NEW selection_keys: ['billable_fixed', 'billable_time', 'non_billable', 'non_billable_project']
+# Nothing to do
+
+sale_timesheet / account.analytic.line    / timesheet_revenue (float)     : NEW
+sale_timesheet / account.invoice          / timesheet_ids (one2many)      : NEW relation: account.analytic.line
+# Done: recomputed by calling _compute_timesheet_revenue method in post
```

Comment:
If you do this recomputation after, then blacklist the initial computation. Check https://github.com/OCA/OpenUpgrade/pull/1421/files#diff-8022026a0884faddafd8988f759b4bb4R14 for seeing how to do it.

Reply:
I'm also worried on how this is going to perform. Maybe an SQL replacement?

Reply:
I think It's very difficult to replace this method:
https://github.com/OCA/OCB/blob/11.0/addons/sale_timesheet/models/account_invoice.py#L54

Reply:
@cubells this is definitively not needed. `timesheet_invoice_type` field is new, so no line will have this value set to `billable_time` or `billable_fixed`, and thus having all to 0 is enough.